src/Collector/RelatedFilesCollector.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Efabrica\PHPStanLatte\Collector;
+
+use Efabrica\PHPStanLatte\Collector\ValueObject\CollectedRelatedFiles;
+use Efabrica\PHPStanLatte\Resolver\CallResolver\CalledClassResolver;
+use Efabrica\PHPStanLatte\Resolver\NameResolver\NameResolver;
+use Efabrica\PHPStanLatte\Type\TypeSerializer;
+use PhpParser\Node;
+use PhpParser\Node\Expr\CallLike;
+use PhpParser\Node\Expr\New_;
+use PHPStan\Analyser\Scope;
+use PHPStan\Node\InClassNode;
+use PHPStan\Reflection\ReflectionProvider;
+
+final class RelatedFilesCollector extends AbstractCollector implements PHPStanLatteCollectorInterface
+{
+    private CalledClassResolver $calledClassResolver;
+
+    private ReflectionProvider $reflectionProvider;
+
+    private NameResolver $nameResolver;
+
+    public function __construct(
+        TypeSerializer $typeSerializer,
+        CalledClassResolver $calledClassResolver,
+        ReflectionProvider $reflectionProvider,
+        NameResolver $nameResolver
+    ) {
+        parent::__construct($typeSerializer);
+        $this->calledClassResolver = $calledClassResolver;
+        $this->reflectionProvider = $reflectionProvider;
+        $this->nameResolver = $nameResolver;
+    }
+
+    public function getNodeType(): string
+    {
+        return Node::class;
+    }
+
+    public function processNode(Node $node, Scope $scope): ?array
+    {
+        $relatedFiles = [];
+        if ($node instanceof InClassNode) {
+            $classReflection = $scope->getClassReflection();
+            if ($classReflection !== null) {
+                foreach ($classReflection->getParents() as $parentClassReflection) {
+                    $relatedFiles[] = $parentClassReflection->getFileName();
+                }
+            }
+        } elseif ($node instanceof New_) {
+            $newClassName = $this->nameResolver->resolve($node->class);
+            if  ($newClassName !== null) {
+                $classReflection = $this->reflectionProvider->getClass($newClassName);
+                if (!$classReflection->isInterface() && !$classReflection->isTrait()) {
+                    $relatedFiles[] = $classReflection->getFileName();
+                }
+            }
+        } elseif ($node instanceof CallLike) {
+            $calledClassName = $this->calledClassResolver->resolve($node, $scope);
+            if ($calledClassName !== null) {
+                $classReflection = $this->reflectionProvider->getClass($calledClassName);
+                if (!$classReflection->isInterface() && !$classReflection->isTrait()) {
+                    $relatedFiles[] = $classReflection->getFileName();
```

Comment:
I think there should be also check that stops searching for additional files on external library classes like what is in MethodCallCollector (https://github.com/efabrica-team/phpstan-latte/blob/1fde35416fdb4e40a88b8ff40c221d180dfceebd/src/Collector/MethodCallCollector.php#L126) to stop unnecessary collection inside Netta/Latte. This might be reason for performance degradation. This check for external class should probably be created as resolver in Resolver\TypeREsolver and used in both MethodCallCollector and RelatedFilesCollector. Also this check might need to be more strict - why do not exclude whole Latte and Nette? Also I thnik we might want to check call arguments - if call is not in same class, do not pass $this and do not pass Template then this call cannot change latte related state so we can ignore it.